ui/desktop/src/contexts/ThemeContext.tsx

```diff
@@ -0,0 +1,161 @@
+    document.documentElement.classList.add('dark');
+    document.documentElement.classList.remove('light');
+  } else {
+    document.documentElement.classList.remove('dark');
+    document.documentElement.classList.add('light');
+  }
+}
+
+interface ThemeProviderProps {
+  children: React.ReactNode;
+}
+
+export function ThemeProvider({ children }: ThemeProviderProps) {
+  const [userThemePreference, setUserThemePreferenceState] =
+    useState<ThemePreference>(loadThemePreference);
+  const [resolvedTheme, setResolvedTheme] = useState<ResolvedTheme>(() =>
+    resolveTheme(loadThemePreference())
+  );
+
+  const setUserThemePreference = useCallback((preference: ThemePreference) => {
+    setUserThemePreferenceState(preference);
+    saveThemePreference(preference);
+
+    const resolved = resolveTheme(preference);
+    setResolvedTheme(resolved);
+    applyThemeToDocument(resolved);
+
+    // Broadcast to other windows via Electron
+    window.electron?.broadcastThemeChange({
+      mode: resolved,
+      useSystemTheme: preference === 'system',
+      theme: resolved,
+    });
+  }, []);
+
+  // Listen for system theme changes when preference is 'system'
+  useEffect(() => {
+    if (userThemePreference !== 'system') return;
+
+    const mediaQuery = window.matchMedia('(prefers-color-scheme: dark)');
+
+    const handleChange = () => {
+      const resolved = getSystemTheme();
+      setResolvedTheme(resolved);
+      applyThemeToDocument(resolved);
+    };
+
+    mediaQuery.addEventListener('change', handleChange);
+    return () => mediaQuery.removeEventListener('change', handleChange);
+  }, [userThemePreference]);
+
+  // Listen for theme changes from other windows (via Electron IPC)
+  useEffect(() => {
+    if (!window.electron) return;
+
+    const handleThemeChanged = (_event: unknown, ...args: unknown[]) => {
+      const themeData = args[0];
+
+      // Validate IPC payload structure
+      if (
+        typeof themeData !== 'object' ||
+        themeData === null ||
+        typeof (themeData as Record<string, unknown>).useSystemTheme !== 'boolean' ||
+        typeof (themeData as Record<string, unknown>).theme !== 'string'
+      ) {
+        console.warn('Invalid theme-changed IPC payload:', themeData);
+        return;
+      }
+
+      const { useSystemTheme, theme } = themeData as { useSystemTheme: boolean; theme: string };
+
+      const newPreference: ThemePreference = useSystemTheme
+        ? 'system'
+        : theme === 'dark'
+          ? 'dark'
+          : 'light';
+
+      setUserThemePreferenceState(newPreference);
+      saveThemePreference(newPreference);
+
+      const resolved = resolveTheme(newPreference);
+      setResolvedTheme(resolved);
+      applyThemeToDocument(resolved);
+    };
+
+    window.electron.on('theme-changed', handleThemeChanged);
+    return () => {
+      window.electron.off('theme-changed', handleThemeChanged);
+    };
+  }, []);
+
+  // Apply theme on initial mount
+  useEffect(() => {
```

Comment:
remove the comment - this is not true, we apply the theme when it changes. also everywhere we call setResolvedThem(...) and then applyThemToDocument(...) remove the second call and just rely on this effect